Example/GiniVision/ComponentAPICoordinator.swift

```diff
@@ -168,151 +206,323 @@ final class ComponentAPICoordinator: NSObject, Coordinator {
-                              animated: Bool) {
-        guard let fromViewController = navigationController
-            .transitionCoordinator?.viewController(forKey: .from) else { return }
-        newDocumentViewController
-            .setNavigationBarHidden(viewController is ComponentAPICameraViewController, animated: true)
+                              animationControllerFor operation: UINavigationControllerOperation,
+                              from fromVC: UIViewController,
+                              to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
         
-        if fromViewController is ComponentAPIReviewViewController &&
-            viewController is ComponentAPICameraViewController {
+        if fromVC is ReviewViewController && operation == .pop {
             reviewScreen = nil
-            documentService.cancelAnalysis()
+            if let document = documentRequests.first?.document {
+                documentService?.delete(document)
+            }
+            documentRequests.removeAll()
         }
         
-        if fromViewController is ComponentAPIAnalysisViewController &&
-            viewController is ComponentAPIReviewViewController {
+        if fromVC is AnalysisViewController && operation == .pop {
+            // Going directly from the analysis to the camera means that
+            // the document is not an image and should be removed
+            if toVC is CameraViewController {
+                documentRequests.removeAll()
+            }
+            
             analysisScreen = nil
-            documentService.cancelAnalysis()
+            documentService?.cancelAnalysis()
         }
         
-        if let resultsScreen = fromViewController as? ResultTableViewController {
-            documentService.sendFeedback(forDocument: resultsScreen.document)
+        if let resultsScreen = fromVC as? ResultTableViewController {
+            documentService?.sendFeedback(with: resultsScreen.result)
         }
+        
+        if let cameraViewController = toVC as? CameraViewController,
+            fromVC is MultipageReviewViewController {
+            cameraViewController
+                .replaceCapturedStackImages(with: documentRequests.compactMap { $0.document.previewImage })
+        }
+        
+        return nil
     }
 }
 
-// MARK: ComponentAPICameraScreenDelegate
+// MARK: - CameraViewControllerDelegate
 
-extension ComponentAPICoordinator: ComponentAPICameraViewControllerDelegate {
-    func componentAPICamera(viewController: UIViewController, didPickDocument document: GiniVisionDocument) {
-        if document.isReviewable {
-            showReviewScreen(withDocument: document)
-        } else {
-            showAnalysisScreen(withDocument: document)
+extension ComponentAPICoordinator: CameraViewControllerDelegate {
+    func camera(_ viewController: CameraViewController, didCapture document: GiniVisionDocument) {
```

Comment:
You extension whitespaces are inconsistent. Here, there's none. In the previous extension you have 1.